ckan/lib/search/__init__.py

```diff
@@ -183,7 +183,9 @@ def rebuild(package_id=None, only_missing=False, force=False, refresh=False,
         else:
             log.info('Rebuilding the whole index...')
             # When refreshing, the index is not previously cleared
-            if not refresh:
+            if (not refresh) and (defer_commit):
```

Comment:
I don't think we need to take into account `defer_commit` when clearing the index. If you want to clear it you'll want to commit the changes otherwise it doesn't have any effect. So these changes should remain unchanged.

Reply:
Thanks.

since I will be removing `-r` and `refresh` , this line should be changed to:
```python
if clear:
  package_index.clear()
```
correct?

Reply:
Correct!